src/Microsoft.AspNet.Authentication/DataHandler/Serializer/TicketSerializer.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNet.Authentication.DataHandler.Serializer
 {
     public class TicketSerializer : IDataSerializer<AuthenticationTicket>
     {
-        private const int FormatVersion = 2;
+        private const int FormatVersion = 3;
```

Comment:
Deserialization would fail in a heterogeneous environment even without this check, the expected formats are different. This check is only to make the failure deterministic and debuggable rather than having random IndexOutOfRangeExceptions or NullReferenceExceptions.